source/draw/gpu/GpuCanvas.ooc

```diff
@@ -18,82 +18,41 @@ use ooc-math
 use ooc-draw
 use ooc-base
 use ooc-collections
-import GpuImage, GpuMap, GpuSurface, GpuContext, GpuYuv420Semiplanar, GpuYuv420Planar
+import GpuImage, GpuMap, GpuSurface, GpuContext, GpuYuv420Semiplanar
 
-GpuCanvas: abstract class extends GpuSurface {
-	_target: GpuImage
-	init: func (=_target, context: GpuContext) { super(this _target size, context, context defaultMap, IntTransform2D identity) }
-	readPixels: virtual func -> ByteBuffer { Debug raise("Trying to read pixels in unimplemented readPixels function"); null }
-	onRecycle: abstract func
-}
-
-GpuCanvasYuv420Semiplanar: class extends GpuCanvas {
-	target ::= this _target as GpuYuv420Semiplanar
+GpuCanvasYuv420Semiplanar: class extends GpuSurface {
+	_target: GpuYuv420Semiplanar
 	transform: FloatTransform3D {
 		get { this _view }
 		set(value) {
 			this _view = this _toLocal * value * this _toLocal
-			this target y canvas _view = this _view
-			this target uv canvas _view = FloatTransform3D createTranslation(-this _view m / 2.0f, -this _view n / 2.0f, -this _view o / 2.0f) * this _view
+			this _target y canvas _view = this _view
+			this _target uv canvas _view = FloatTransform3D createTranslation(-this _view m / 2.0f, -this _view n / 2.0f, -this _view o / 2.0f) * this _view
```

Comment:
No, that operator is not defined for FloatTransform3D and even if it was it would not be correct.